### DISCUSSION TITLE
change color of listchars

### DISCUSSION THREAD

**Discussion by john-soda** [lukas-reineke/indent-blankline.nvim#334] [Q&A]
I use only the end of line symbol 
`vim.opt.listchars:append("eol:↴")`
to recognize trailing whitespace but I don't like the color of it because I barely can see it and I don't want to change my theme. 
Is there a way to do it?
Here my whole current config.

```
vim.opt.termguicolors = true
vim.opt.list = true
vim.opt.listchars:append("eol:↴")
vim.cmd [[highlight IndentBlanklineIndent guifg=#999999 gui=nocombine]]

require("indent_blankline").setup {
    show_current_context = true,
    show_current_context_start = true,
    char_highlight_list = {
        "IndentBlanklineIndent",
    }
}
```
edit: can I change the highlight color of the show_current_context, too?

**Comment by lukas-reineke** [comment]
There is currently no dedicated highlight group for the eol character. But you can change the color with `:help hl-IndentBlanklineChar`
For the context there is `:help hl-IndentBlanklineContextChar` and `:help hl-IndentBlanklineContextStart`

**Reply by john-soda** [reply]
Means this the color of eol character can't be changed, at all? Or it cannot be changed solely? Because I tried everything and it doesn't change.
Thanks very much for your help!

edit: is it possible to make the IndentBlanklineContextChar Bold?

**Reply by lukas-reineke** [reply]
> Means this the color of eol character can't be changed, at all? Or it cannot be changed solely?

You can change it with `IndentBlanklineChar`

> is it possible to make the IndentBlanklineContextChar Bold?

Yes it is, take a look at `:help bold` and `:help highlight-gui`

**Reply by john-soda** [reply]
Thanks very much for your help and for indent-blankline at all!
I tried IndentBlanklineChar before, but my eol sign never changes the color. I tried many settings and I was able to change the color of many things. But not for the the eol or whitspace_fg elements at all. The utils.lua file in line 214 is responsible for the default color of  IndentBlanklineChar. I read the docs twice. 
I think I am too stupid to understand how to change the color of eol. Attached is my current config as test and pic.
```lua
vim.opt.termguicolors = true
vim.cmd [[highlight IndentBlanklineIndent1 guifg=#999999 gui=nocombine]]
vim.cmd [[highlight IndentBlanklineIndent2 guifg=#625e5b gui=nocombine]]
vim.cmd [[highlight IndentBlanklineChar guifg=#00FF00 gui=nocombine]]
-- show_current_context color in indent
vim.cmd [[highlight IndentBlanklineContextChar guifg=#d29fd3 gui=nocombine]]

vim.opt.list = true
vim.opt.listchars:append("eol:↴")
-- vim.opt.listchars:append("space:⋅")

require("indent_blankline").setup {
    space_char_blankline = " ",
    show_end_of_line = true,
    show_current_context = true,
    show_current_context_start = true,
    
    char_highlight_list = {
        "IndentBlanklineIndent1",
        "IndentBlanklineIndent2",
    },
    indent_blankline_use_treesitter = true,
}
```
I use neovim with the colorschem: default

![Screenshot from 2021-12-13 08-48-04](https://user-images.githubusercontent.com/11722816/145775343-8179724f-5188-400d-b1f7-14beb82ede53.png)

**Reply by lukas-reineke** [reply]
Oh do you mean the eol character at the end of normal lines with content?
Those have nothing to do with indent-blankline.nvim. That's a build in feature.

The highlight group for those is `:help hl-NonText`

**Reply by john-soda** [reply]
OK thanks for clarifying. But what did you mean?
And Another thing, about getting the IndentBlanklineContextChar in bold.
`vim.cmd [[highlight IndentBlanklineContextChar guifg=#d29fd3 gui=bold,nocombine]]`
did not make it bold, I can change the color but not the boldness.

**Reply by john-soda** [reply]
hl-NonText worked! thanks for all the hints!
Do you have a last hint to get IndentBlanklineContextChar bold?

**Reply by lukas-reineke** [reply]
The command is correct. If it is not getting bold, it is probably because your terminal doesn't support it, or is not setup correctly.

**Reply by john-soda** [reply]
Thanks, for all the help!
